django/db/backends/oracle/compiler.py

```diff
@@ -42,7 +42,36 @@ def as_sql(self, with_limits=True, with_col_aliases=False, subquery=False):
 
 
 class SQLInsertCompiler(compiler.SQLInsertCompiler, SQLCompiler):
-    pass
+    def as_sql(self):
```

Comment:
Might it be possible to use features flags, etc. in `SQLCompiler.as_sql()` (`django/db/models/sq/compiler.py`) so we don't need to duplicate portions of that method here?